schema/v1-example-statement.json

```diff
@@ -0,0 +1,36 @@
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "subject": [
+    {
+      "name": "oak_functions_loader",
+      "digest": {
+        "sha256": "d4d5899a3868fbb6ae1856c3e55a32ce35913de3956d1973caccd37bd0174fa2"
+      }
+    }
+  ],
+  "predicateType": "https://slsa.dev/provenance/v0.2",
+  "predicate": {
+    "builder": {
+      "id": "https://github.com/project-oak/transparent-release"
+    },
+    "buildType": "https://github.com/project-oak/transparent-release/amber-slsa-buildtype/v1.json",
+    "buildConfig": {
+      "command": "./scripts/runner build-functions-server",
+      "outputPath": "./oak_functions/loader/bin/oak_functions_loader"
+    },
+    "materials": [
+      {
+        "uri": "gcr.io/oak-ci/oak@sha256:53ca44b5889e2265c3ae9e542d7097b7de12ea4c6a33785da8478c7333b9a320",
+        "digest": {
+          "sha256": "15dc16c42a4ac9ed77f337a4a3065a63e444c29c18c8cf69d6a6b4ae678dca5c"
+        }
+      },
+      {
+        "uri": "https://github.com/project-oak/oak",
```

Comment:
Is this actually how the commit is referred to in the build? That seems to be the definition of this field in https://slsa.dev/provenance/v0.2 (which I find somewhat confusing myself, since it seems to assume that the URI is stable / immutable (which here is not, obviously). Or is there a URI format for git commits that includes the hash of the commit, similar to how the Docker URI includes the digest of the image. Anyways please don't block on this, it may be something that we can bring up with the SLSA people.

Also unfortunately my usual suggestion of using https://github.com/google/ent does not apply here, since git still uses sha1, and Ent only supports sha256 (so far) :sweat:

Reply:
At the moment it seems this is the way, yeah. I also wondered about a URI format for cloning specific git commits, but that doesn't seem to be a thing. 

There is an in-toto issue about differentiating between different "kinds" of digests that might improve this: https://github.com/in-toto/attestation/issues/28

As an aside, you probs know this already, but git does plan to move to sha256, but afaik it's taking a while as it's part of larger refactor to make git hash function agnostic.